### PR TITLE
Use bcryptjs for Alpine Docker images

### LINKED ISSUE #719: Docker images have a segmentation fault problem with bcrypt
Alpine linux based images use musl libc instead of the standard glibc and thus are not compatible with compiled binaries. node-pre-gyp currently does not test the pre-compiled binaries to be ABI-compatible and thus you may see segfaults.

Use multi stage build to recompile the library and then copy the binares in a brand new image

### PR DESCRIPTION
**Pull request by XVincentX**
Connect #719 
Closes #719

### CODE REVIEW

**Comment by XVincentX** [comment]
@kevinswiber Sure, we can definitely use `bcryptjs` and make the day.

Just one thing more: this is not complicating the flow for the final user. They'll keep downloading the ready to use image but I agree that this is complicated. I'll modify the PR.

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ ENV CHOKIDAR_USEPOLLING true
 
 VOLUME /var/lib/eg
 
-RUN yarn global add express-gateway@$EG_VERSION
+RUN yarn global add express-gateway@$EG_VERSION --ignore-optional
```
Comment (kevinswiber) [inline Dockerfile]: I asked this before, but I'm not sure I got a response.  Why are we using `yarn` instead of `npm`?

File: Dockerfile-master [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:8-alpine
+FROM node:8
```
Comment (kevinswiber) [inline Dockerfile-master]: Why are we not using the Alpine distribution for master builds?

**Comment by kevinswiber** [comment]
> To be honest here I was wondering whether wouldn't it be better to simply drop bcrypt and have bcryptjs only. It is not a speed-critical part of the system (user creation) and it would remove a lot of hassle here.

@XVincentX Can you put it in a new issue? We can close this one out.  Thanks!

**Comment by XVincentX** [comment]
https://github.com/ExpressGateway/express-gateway/issues/724

File: Dockerfile (hunk 1 above)
Comment (kevinswiber) [reply Dockerfile]: Ah, okay.  You can change the global path in NPM, though.  `npm config set prefix "/chicken/noodle/soup"`

**Comment by XVincentX** [comment]
I do not really care that much about the package manager — but at the time it was not working with NPM. It's surely because I was missing something with Docker Users and permissions, but with yarn it was working out of the box, it's included in the Node Image so I went in this direction.

I'm ok revisiting it later!

**Pull request merged**